### PR TITLE
OTT-1225: Tracking-Beacon-First (TBF) for banner creatives

### PR DESCRIPTION
**Pull request by pm-nilesh-chate**
# Description

Please add change description or link to ticket, docs, etc.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: modules/pubmatic/openwrap/database/mysql/tracking_beacon_first.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Package database provides functionalities to interact with the giym database.
```
Comment (ashishshinde-pubm) [inline modules/pubmatic/openwrap/database/mysql/tracking_beacon_first.go:1]: minor comment - package name mismatch

File: modules/pubmatic/openwrap/cache/gocache/tracking_beacon_first.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Package dbcache contains caching functionalities of header-bidding repostiry
```
Comment (ashishshinde-pubm) [inline modules/pubmatic/openwrap/cache/gocache/tracking_beacon_first.go:1]: minor comment - gocache package name

**Review by ashishshinde-pubm — APPROVED** [review]
LGTM
added minor comments.

File: modules/pubmatic/openwrap/database/mysql/tracking_beacon_first.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+		if err := rows.Scan(&pubID, &trafficDetails); err != nil {
+			glog.Error("ErrRowScanFailed GetTBFRateQuery pubid: ", pubID, " err: ", err.Error())
+			continue
+		}
+
+		// convert trafficDetails into map[profileId]traffic
+		var profileTrafficRate map[int]int
+		if err := json.Unmarshal([]byte(trafficDetails), &profileTrafficRate); err != nil {
+			glog.Error("ErrJSONUnmarshalFailed TBFProfileTrafficRate pubid: ", pubID, " trafficDetails: ", trafficDetails, " err: ", err.Error())
```
[the remark this answers is not in the document]
Comment (pm-nilesh-chate) [reply modules/pubmatic/openwrap/database/mysql/tracking_beacon_first.go:34]: pbs modules logging is still pending, will do that once db package is migrated

**Review by pm-aadit-patil — APPROVED** [review]

**Review by ashishshinde-pubm — APPROVED** [review]

**Pull request merged**